gpytorch/test/lazy_tensor_test_case.py

```diff
@@ -405,6 +405,58 @@ def test_add_diag(self):
                 actual[..., i, i] = actual[..., i, i] + other_diag[..., i]
             self.assertAllClose(res, actual, **self.tolerances["diag"])
 
+    def test_add_low_rank(self):
+        lazy_tensor = self.create_lazy_tensor()
+        lazy_tensor = self.create_lazy_tensor()
+        evaluated = self.evaluate_lazy_tensor(lazy_tensor)
+        new_rows = torch.randn(*lazy_tensor.shape[:-1], 3)
+
+        summed_lt = evaluated + new_rows.matmul(new_rows.transpose(-1, -2))
+        new_lt = lazy_tensor.add_low_rank(new_rows)
+
+        # check that the concatenation is okay
+        self.assertAllClose(new_lt.evaluate(), summed_lt)
+
+        # check that the root approximation is close
+        rhs = torch.randn(lazy_tensor.size(-1))
+        summed_rhs = summed_lt.matmul(rhs)
+        root_rhs = new_lt.root_decomposition().matmul(rhs)
+        self.assertAllClose(root_rhs, summed_rhs, **self.tolerances["root_decomposition"])
+
+        # check that the inverse root decomposition is close
+        summed_solve = torch.solve(rhs.unsqueeze(-1), summed_lt)[0].squeeze(-1)
+        root_inv_solve = new_lt.root_inv_decomposition().matmul(rhs)
+        self.assertAllClose(root_inv_solve, summed_solve, **self.tolerances["root_inv_decomposition"])
+
+    def test_cat_rows(self):
```

Comment:
I want to be sure that this test case is capturing all the branching logic. Just to be clear, one of the sub LT classes will go into the `if l_is_triang:` blocks, right?

Reply:
Yes, most of the lazy tests are going to hit `l_is_triang` but enforcing an error in the false case hits at least constant_mul_lazy_tensor (and probably others that have specialized roots).